### DISCUSSION TITLE
Help trying to reproduce some packets captured with Wireshark

### DISCUSSION THREAD

**Discussion by alexpdp7** [libusb/hidapi#679] [Q&A] [answered]
Hi,

Sorry, I'm a bit lost. I'm trying to replay some interaction with a USB device (a tablet keyboard) that I have captured using Wireshark. Surprisingly, I'm not able to export the human-readable description of the package in text, so sorry for the screenshot:

![image](https://github.com/libusb/hidapi/assets/1150630/fb328eb4-874e-4e70-80f2-2cca51ffdc4f)

This is the action I'm trying to replay (it's switching the ctrl/fn keys on the tablet keyboard). This is a toggle; the variation between successive toggles is the `IRP ID` (which I suspect is not important) and the data fragment (`09b401` or `09b402`).

I found <https://github.com/todbot/hidapitester>, which seems like a handy tool to turn that packet into a command and send it, but I'm struggling to find out what parameters to use.

I think I want something like:

```
hidapitester [...] --send-output 0x09,0x09,0xb4,0x02
```

, but I'm struggling to find the right combination.

Help?

Álex

**Comment by manueljaeckle** [comment, marked as answer]
Hi, I stumbled upon this and I'm not sure if it's still relevant for you or if you've already found a solution. I had the same problem (switching the Fn key on Linux on the ThinkPad X12) and quickly documented how I solved it in a repo:

https://github.com/manueljaeckle/thinkpad_x12_fn_switcher

Just dropping this here, maybe it will be useful for someone.

**Reply by alexpdp7** [reply]
I had not found a solution! Thank you so much. I can confirm it worked for me.
